File: app/src/main/java/com/example/client/app/RxBus.java
package com.example.client.app;

import com.example.client.models.event.Event;

import io.reactivex.subjects.PublishSubject;


public class RxBus {

    private static PublishSubject<Event> mInstance;

    public static PublishSubject<Event> getInstance() {

        if (mInstance == null) {
            mInstance = PublishSubject.create();
        }
        return mInstance;
    }
}

File: app/src/main/java/com/example/client/app/RxBus.kt
package com.example.client.app

import com.example.client.models.event.Event
import io.reactivex.subjects.PublishSubject

object RxBus {
    private var mInstance: PublishSubject<Event>? = null
    fun newInstance(): PublishSubject<Event> {
        if (mInstance == null) {
            mInstance = PublishSubject.create()
        }
        return mInstance!!
    }
}